neuro_san/service/mcp/processors/mcp_tools_processor.py

```diff
@@ -166,54 +176,78 @@ async def build_tool_call_result(
         }
         # Construct actual tool call result:
         if result_structure is not None:
-            call_result["result"]["structuredContent"] = result_structure
+            call_result["result"]["structuredContent"] = {}
+            call_result["result"]["structuredContent"]["result"] = result_structure
             # For backward compatibility, also add text version of structure:
             structure_str: str = f"```json\n{json.dumps(result_structure, indent=2)}\n```"
             result_text = result_text + structure_str
+        if result_sly_data is not None:
+            call_result["result"]["structuredContent"]["sly_data"] = result_sly_data
         call_result["result"]["content"][0]["text"] = RequestsUtil.safe_message(result_text)
```

Comment:
Per MCP spec recommendation, all structured data - sly data and maybe chat response structured part -
should go into "structuredContent" part of tool call result.